pkg/config/middleware_test.go

```diff
@@ -100,6 +117,120 @@ func TestAuthMiddleware(t *testing.T) {
+	})
+
+	t.Run("it will redirect if jwt enabled with invalid cookie token and valid header token", func(t *testing.T) {
+		Config.JWTAuthEnabled = true
+		defer func() { Config.JWTAuthEnabled = false }()
+		hh := SetupGlobalMiddleware(h)
+
+		res := httptest.NewRecorder()
+		res.Body = new(bytes.Buffer)
+		req, _ := http.NewRequest("GET", "http://localhost:18000/api/v1/flags", nil)
+		req.AddCookie(&http.Cookie{
+			Name:  "access_token",
+			Value: "invalid_jwt",
+		})
+		req.Header.Add("Authorization", "Bearer "+validHS256JWTToken)
+		hh.ServeHTTP(res, req)
+		assert.Equal(t, http.StatusTemporaryRedirect, res.Code)
+	})
+
+	t.Run("it will redirect if jwt enabled and a cookie token encrypted with the wrong method", func(t *testing.T) {
+		Config.JWTAuthEnabled = true
+		Config.JWTAuthSigningMethod = "RS256"
+		defer func() {
+			Config.JWTAuthEnabled = false
+			Config.JWTAuthSigningMethod = "HS256"
+		}()
+		hh := SetupGlobalMiddleware(h)
+
+		res := httptest.NewRecorder()
+		res.Body = new(bytes.Buffer)
+		req, _ := http.NewRequest("GET", "http://localhost:18000/api/v1/flags", nil)
+		req.AddCookie(&http.Cookie{
+			Name:  "access_token",
+			Value: "invalid_jwt",
+		})
+		hh.ServeHTTP(res, req)
+		assert.Equal(t, http.StatusTemporaryRedirect, res.Code)
+	})
+
+	t.Run("it will pass if jwt enabled with correct header token encrypted using RS256", func(t *testing.T) {
+		Config.JWTAuthEnabled = true
+		Config.JWTAuthSigningMethod = "RS256"
+		Config.JWTAuthSecret = `-----BEGIN PUBLIC KEY-----
+MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDdlatRjRjogo3WojgGHFHYLugd
+UWAY9iR3fy4arWNA1KoS8kVw33cJibXr8bvwUAUparCwlvdbH6dvEOfou0/gCFQs
+HUfQrSDv+MuSUMAe8jzKE4qW+jK+xQU9a03GUnKHkkle+Q0pX/g6jXZ7r1/xAK5D
+o2kQ+X5xK9cipRgEKwIDAQAB
+-----END PUBLIC KEY-----`
+		defer func() {
+			Config.JWTAuthEnabled = false
+			Config.JWTAuthSigningMethod = "HS256"
+			Config.JWTAuthSecret = ""
+		}()
+		hh := SetupGlobalMiddleware(h)
+
+		res := httptest.NewRecorder()
+		res.Body = new(bytes.Buffer)
+		req, _ := http.NewRequest("GET", "http://localhost:18000/api/v1/flags", nil)
+		req.Header.Add("Authorization", "Bearer "+validRS256JWTToken)
+		hh.ServeHTTP(res, req)
+		assert.Equal(t, http.StatusOK, res.Code)
+	})
+
+	t.Run("it will pass if jwt enabled with valid cookie token with passphrase", func(t *testing.T) {
+		Config.JWTAuthEnabled = true
+		Config.JWTAuthSecret = "mysecret"
+		defer func() {
+			Config.JWTAuthEnabled = false
+			Config.JWTAuthSecret = ""
+		}()
+		hh := SetupGlobalMiddleware(h)
+
+		res := httptest.NewRecorder()
+		res.Body = new(bytes.Buffer)
+		req, _ := http.NewRequest("GET", "http://localhost:18000/api/v1/flags", nil)
+		req.AddCookie(&http.Cookie{
+			Name:  "access_token",
+			Value: validHS256JWTTokenWithSecret,
+		})
+		hh.ServeHTTP(res, req)
+		assert.Equal(t, http.StatusOK, res.Code)
+	})
+
+	t.Run("it will pass if jwt enabled with correct HS256 token cookie and signing method is wrong", func(t *testing.T) {
```

Comment:
nit: change to "it will pass with a correct HS256 token cookie when signing method is wrong and it defaults to empty string secret"